### ISSUE TITLE
Image config is not included when pulling an image by hash from DockerHub

### ISSUE THREAD

**Issue by emi80** [apptainer/singularity#1738]
## Version of Singularity:

2.5.2-dist

### Expected behavior

The Docker image config should be fetched and included in the resulting Singularity image.

### Actual behavior

The config object of the Docker image is not fetched and the pull command log says at some point:
```
...
WARNING No metadata will be included.
...
```

The config is included pulling the image by the corresponding tag (see below).

The issue is due to the fact that the Docker image manifest uses [schema version 2](https://docs.docker.com/registry/spec/manifest-v2-2/#image-manifest-field-descriptions) when refferring to it by hash and fetching a separate config object is not currently supported [here](https://github.com/singularityware/singularity/blob/master/libexec/python/docker/api.py#L539).

### Steps to reproduce behavior

```
$ singularity pull docker://grape/mapping@sha256:5c0b8b7796fbbe6d7e7d2d5d15796ded91fa9ff60c7f6cb915f0e1193a7fca4a

$ singularity exec mapping\@sha256-5c0b8b7796fbbe6d7e7d2d5d15796ded91fa9ff60c7f6cb915f0e1193a7fca4a.simg bash -c 'echo $PATH'
/bin:/sbin:/usr/bin:/usr/sbin:/usr/local/bin:/usr/local/sbin
```
 vs

```
$ singularity pull docker://grape/mapping:gem-1.7.1

$ singularity exec mapping-gem-1.7.1.simg bash -c 'echo $PATH'
/usr/local/lib/python2.7/dist-packages/gem/gembinaries/:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
```

**Comment by jscook2345** [comment]
@emi80 

This appears to be fixed in version `3.1.0`. Do you want us to take a look at the `2.X` versions and see if it is fixed there as well?

**Comment by jscook2345** [comment]
This is fixed in the latest version. Please comment here if you have further questions. Thanks!
